plugins/cilium-cni/lib/deletion_queue.go

```diff
@@ -110,35 +114,54 @@ func (dc *DeletionFallbackClient) tryQueueLock() error {
 	if err != nil {
 		return fmt.Errorf("failed to acquire lock: %w", err)
 	}
+
+	dc.logger.Debug("Deletion Queue lock acquired",
+		logfields.Path, defaults.DeleteQueueLockfile,
+		logfields.Duration, time.Since(startTime))
 	dc.lockfile = lf
 	return nil
 }
 
-// EndpointDelete deletes an endpoint given by an endpoint id, either
-// by directly accessing the API or dropping in a queued-deletion file.
-// endpoint-id is a qualified endpoint reference, e.g. "container-id:XXXXXXX"
-func (dc *DeletionFallbackClient) EndpointDelete(id string) error {
-	if dc.cli != nil {
-		return dc.cli.EndpointDelete(id)
-	}
-
-	// fall-back mode
-	if dc.lockfile != nil {
-		dc.logger.Info(
-			"Queueing deletion request for endpoint",
-			logfields.EndpointID, id,
-		)
-		return dc.enqueueDeletionRequestLocked(id)
-	}
-
-	return errors.New("attempt to delete with no valid connection")
-}
-
 // EndpointDeleteMany deletes multiple endpoints based on the endpoint deletion request,
 // either by directly accessing the API or dropping in a queued-deletion file.
 func (dc *DeletionFallbackClient) EndpointDeleteMany(req *models.EndpointBatchDeleteRequest) error {
 	if dc.cli != nil {
-		return dc.cli.EndpointDeleteMany(req)
+		startTime := time.Now()
+		err := dc.cli.EndpointDeleteMany(req)
+		if err == nil {
+			dc.logger.Debug("Endpoint delete request successful",
+				logfields.Request, req,
+				logfields.Duration, time.Since(startTime),
+			)
+			return nil
+		}
+
+		status, ok := err.(runtime.ClientResponseStatus)
+		if !ok || !status.IsCode(http.StatusServiceUnavailable) {
+			// Propagate unhandled Endpoint delete errors.
+			return err
+		}
+
+		// Fallback to queuing endpoint delete if cilium Endpoint service is unavailable.
+		// Once ready, cilium-agent will process all the pending deletes in this queue.
+		dc.logger.Warn(
```

Comment:
If this is an expected condition that the user can't do anything about, I would suggest either Debug or at most, Info.

```suggestion
		dc.logger.Debug(
```

As a separate note, we could decide to always emit an Info level log message when queueing deletion for easier tracing of the CNI flow.

Reply:
We do have an info log as part of `enqueueDeletionRequestLocked` method.